symbol_table/src/tests.rs

```diff
@@ -205,3 +224,78 @@ fn keep_vars() {
         assert_eq!(table.get(var), Some(&data));
     }
 }
+
+#[test]
+fn newing_stores_a_copy() {
+    let mut table = Table::new();
+    let var = var_m("var", &[]);
+
+    let level_zero = "zero".try_into().unwrap();
+    table.set(&var, &level_zero).unwrap();
+
+    table.push_new_frame();
+    table.new_var(&[&var.name]);
+    let level_one = "one".try_into().unwrap();
+    table.set(&var, &level_one).unwrap();
+
+    table.push_new_frame();
+    table.new_var(&[&var.name]);
+    let level_two = "two".try_into().unwrap();
+    table.set(&var, &level_two).unwrap();
+
+    table.push_new_frame();
+    table.new_var(&[&var.name]);
+    assert_eq!(table.get(&var), None);
+
+    table.pop_new_frame();
+    assert_eq!(table.get(&var), Some(&level_two));
+
+    table.pop_new_frame();
+    assert_eq!(table.get(&var), Some(&level_one));
+
+    table.pop_new_frame();
+    assert_eq!(table.get(&var), Some(&level_zero));
+}
+
+#[test]
+fn non_newed_variables_are_still_accessible() {
```

Comment:
Rename to assumed variables are accessible. 
Try to avoid non-something if there is a better name for that case.